assets/src/edit-story/components/movable-element/index.js

```diff
@@ -0,0 +1,156 @@
+	} = useStory();
+
+	const frame = {
+		translate: [ 0, 0 ],
+		rotate: rotationAngle,
+	};
+
+	const setStyle = ( target ) => {
+		target.style.transform = `translate(${ frame.translate[ 0 ] }px, ${ frame.translate[ 1 ] }px) rotate(${ frame.rotate }deg)`;
+	};
+
+	const resetMoveable = ( target ) => {
+		frame.translate = [ 0, 0 ];
+		setStyle( target );
+		if ( moveable.current ) {
+			moveable.current.updateRect();
+		}
+	};
+
+	const handleSelectElement = useCallback( ( elId, evt ) => {
+		if ( evt.metaKey ) {
+			toggleElementIdInSelection( elId );
+		} else {
+			selectElementById( elId );
+		}
+		evt.stopPropagation();
+	}, [ toggleElementIdInSelection, selectElementById ] );
+
+	const comp = getComponentForType( type );
+	const Comp = comp; // why u do dis, eslint?
+
+	return (
+		<>
+			<Element
+				key={ id }
+				onClick={ ( evt ) => handleSelectElement( id, evt ) }
+			>
+				<Comp { ...rest } forwardedRef={ setTargetEl } />
+			</Element>
+			{ targetEl && <Moveable
+				ref={ moveable }
+				target={ targetEl }
+				draggable={ true }
+				resizable={ selected }
+				rotatable={ selected }
+				onDrag={ ( { target, beforeTranslate } ) => {
+					frame.translate = beforeTranslate;
+					setStyle( target );
+				} }
+				onDragStart={ ( { set } ) => {
+					set( frame.translate );
+				} }
+				onDragEnd={ ( { target } ) => {
+					const newProps = { x: x + frame.translate[ 0 ], y: y + frame.translate[ 1 ] };
+					if ( ! selected ) {
+						// Multi-selection is always selected previously,
+						// we can use just setting single element properties here.
+						selectElementById( id );
+						setPropertiesById( id, newProps );
+					} else {
+						setPropertiesOnSelectedElements( newProps );
+					}
+					resetMoveable( target );
+				} }
+				onResizeStart={ ( { setOrigin, dragStart } ) => {
+					setOrigin( [ '%', '%' ] );
+					if ( dragStart ) {
+						dragStart.set( frame.translate );
+					}
+				} }
+				onResize={ ( { target, width, height, drag } ) => {
+					target.style.width = `${ width }px`;
+					target.style.height = `${ height }px`;
+					frame.translate = drag.beforeTranslate;
+					setStyle( target );
+				} }
+				onResizeEnd={ ( { target } ) => {
+					setPropertiesOnSelectedElements( {
+						width: parseInt( target.style.width ),
+						height: parseInt( target.style.height ),
+						x: x + frame.translate[ 0 ],
+						y: y + frame.translate[ 1 ],
+					} );
+					resetMoveable( target );
+				} }
+				onRotateStart={ ( { set } ) => {
+					set( frame.rotate );
+				} }
+				onRotate={ ( { target, beforeRotate } ) => {
+					frame.rotate = beforeRotate;
+					setStyle( target );
+				} }
+				onRotateEnd={ () => {
+					setPropertiesOnSelectedElements( { rotationAngle: frame.rotate } );
+				} }
+				origin={ false }
+				pinchable={ true }
+				keepRatio={ 'image' === type }
+				renderDirections={ 'image' === type ? [ 'nw', 'ne', 'sw', 'se' ] : [ 'n', 's', 'e', 'w', 'nw', 'ne', 'sw', 'se' ] }
```

Comment:
This may get more complex as we add new elements. Maybe this is better as a function?

Reply:
I think this is mainly a property such as `preserveAspectRatio=true|false`.